bson/src/main/org/bson/BsonBinary.java

```diff
@@ -75,6 +78,69 @@ public BsonBinary(final byte type, final byte[] data) {
         this.data = data;
     }
 
+    /**
+     * Construct a Type 4 BsonBinary from the given UUID.
+     *
+     * @param uuid the UUID
+     */
+    public BsonBinary(final UUID uuid) {
+        this(uuid, UuidRepresentation.STANDARD);
+    }
+
+    /**
+     * Construct a new instance from the given UUID and UuidRepresentation
+     *
+     * @param uuid the UUID
+     * @param uuidRepresentation the UUID representation
+     */
+    public BsonBinary(final UUID uuid, final UuidRepresentation uuidRepresentation) {
+        if (uuid == null) {
+            throw new IllegalArgumentException("uuid may not be null");
+        }
+        if (uuidRepresentation == null) {
+            throw new IllegalArgumentException("uuidRepresentation may not be null");
+        }
+        this.data = UuidHelper.encodeUuidToBinary(uuid, uuidRepresentation);
+        this.type = uuidRepresentation == UuidRepresentation.STANDARD
+                ? BsonBinarySubType.UUID_STANDARD.getValue()
+                : BsonBinarySubType.UUID_LEGACY.getValue();
+    }
+
+    /**
+     * Returns the binary as a UUID. The binary type must be 4.
+     *
+     * @return the uuid
+     */
+    public UUID asUuid() {
+        if (type != BsonBinarySubType.UUID_STANDARD.getValue()) {
+            throw new BsonInvalidOperationException("BsonBinarySubType must be UUID_STANDARD.");
+        }
+
+        return UuidHelper.decodeBinaryToUuid(this.data.clone(), this.type, UuidRepresentation.STANDARD);
```

Comment:
Minor stylistic note, you don't need to use `this.<variable name>` if there is no local `<variable name>` variable.  So `data.clone()` and `type` are all that's needed here and in the next method.